src/test/java/nl/tudelft/ewi/devhub/server/backend/warnings/LargeCommitWarningGeneratorTest.java

```diff
@@ -0,0 +1,137 @@
+        when(commit.getCommitId()).thenReturn(COMMIT_ID);
+        when(commit.isMerge()).thenReturn(false);
+
+        when(abstractCommitWarningGenerator.getRepository(commit)).thenReturn(repositoryApi);
+        when(abstractCommitWarningGenerator.getGitCommit(commit)).thenReturn(commitApi);
+
+        /* Build the mocked repo */
+        when(repositoriesApi.getRepository(REPOSITORY_NAME)).thenReturn(repositoryApi);
+        when(repositoryApi.getCommit(COMMIT_ID)).thenReturn(commitApi);
+        when(commitApi.diff()).thenReturn(diffModel);
+        when(diffModel.getDiffs()).thenReturn(diffs);
+
+        /* Used to prevent NullPointerExceptions in tooManyLineChanges */
+        when(diffs.stream()).thenReturn(stream);
+        when(stream.filter(anyObject())).thenReturn(stream);
+        when(stream.flatMap(anyObject())).thenReturn(objectStream);
+        when(objectStream.flatMap(anyObject())).thenReturn(objectStream);
+        when(objectStream.filter(anyObject())).thenReturn(objectStream);
+
+        /* Set up a warning, used to test if the generator returns an equal object */
+        LargeCommitWarning warning = new LargeCommitWarning();
+        warning.setCommit(commit);
+        testEquals = Sets.newHashSet(warning);
+    }
+
+    /**
+     * Confirms that the warning is not generated when the commit is a merge.
+     */
+    @Test
+    public void testIsMerge() {
+        when(commit.isMerge()).thenReturn(true);
+        Set<LargeCommitWarning> empty = generator.generateWarnings(commit, gitPush);
+        assertTrue(empty.isEmpty());
+    }
+
+    /**
+     * Confirms that the warning is generated when there are too many files committed.
+     */
+    @Test
+    public void testTooManyFiles() {
+        when(diffs.size()).thenReturn(1);
+        when(configurable.getIntegerProperty(anyString(), anyString())).thenReturn(0);
+
+        Set<LargeCommitWarning> warnings = generator.generateWarnings(commit, gitPush);
+
+        verify(diffs).size();
+
+        assertFalse(warnings.isEmpty());
+        assertEquals(1, warnings.size());
+        assertEquals(testEquals, warnings);
+    }
+
+    /**
+     * Confirms that the warning is generated when there are too many lines committed.
+     */
+    @Test
+    public void testTooManyLineChanges() {
+        when(objectStream.count()).thenReturn((long) 1);
+        when(configurable.getIntegerProperty(anyString(), anyString())).thenReturn(0);
+
+        Set<LargeCommitWarning> warnings = generator.generateWarnings(commit, gitPush);
+
+        assertFalse(warnings.isEmpty());
+        assertEquals(1, warnings.size());
+        assertEquals(testEquals, warnings);
+    }
+
+    /**
+     * Confirms that no warning is generated when the commit is deemed good.
+     */
+    @Test
+    public void testGoodCommit() {
+        Set<LargeCommitWarning> warnings = generator.generateWarnings(commit, gitPush);
+
+        assertTrue(warnings.isEmpty());
```

Comment:
[`assertThat(warnings, is(empty()))`](http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/Matchers.html#empty()) is easier to read and provides a better error message when the test fails.